w08h03/test/pgdp/datastructures/UnitTest.java

```diff
@@ -14,48 +18,88 @@
-	}
+	static int seed = 69420;
 
-	@ParameterizedTest
-	@DisplayName("Dynamic Test")
-	@MethodSource
-	void dynamicTests(String type, int[] expected, int[] input) {
-		QuarternarySearchTree<Integer> tree = new QuarternarySearchTree<Integer>();
+	<Type extends Comparable<Type>> void testArray(Type[] input) {
+		QuarternarySearchTree<Type> tree = new QuarternarySearchTree<Type>();
+		Type[] expected = Arrays.copyOf(input, input.length);
+		Arrays.sort(expected);
 
-		for (int element : input) {
+		for (Type element : input) {
 			tree.insert(element);
 		}
 
 		int position = 0;
-		for (int element : tree) {
-			int exp = expected[position++];
-			assertEquals(exp, element, type + ": Invalid Output at position [" + (position-1) + "]: Expected [" + exp + "], got [" + element + "]");
+		for (Type element : tree) {
+			Type exp = expected[position++];
+			assertEquals(exp, element, "Invalid Output at position [" + (position - 1) + "]: Expected [" + exp
+					+ "], got [" + element + "]");
 		}
 
-		assertEquals(expected.length, position, type + ": Invalid Iteration Count. Expected [" + expected.length + "] got [" + position + "]");
+		assertEquals(expected.length, position,
+				"Invalid Iteration Count. Expected [" + expected.length + "] got [" + position + "]");
+	}
+
+	@Test
+	@DisplayName("should iterate over empty graph")
+	public void emptyTest() {
+		assertThrows(NoSuchElementException.class, () -> (new QuarternarySearchTree<Integer>()).iterator().next());
+	}
+
+	@Test
+	@DisplayName("should iterate over graph in artemis example")
+	public void artemisTest() {
+		testArray(new Integer[] { 8, 4, 12, 1, 5, 9, 13, 3, 7, 11, 15, 2, 6, 10, 14 });
+	}
+
+	@Test
+	@DisplayName("should iterate over a graph of one repeat element")
+	public void oneElementTest() {
+		testArray(new Integer[] { 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 });
+	}
+
+	@Test
+	@DisplayName("should iterate over a numeric graph")
+	public void numericTest() {
+		IntStream.range(0, 10).forEach(i -> {
+			var args = new Integer[i];
+			var rnd = new Random(seed + i);
+
+			for (int j = 0; j < args.length; j++) {
+				args[j] = rnd.nextInt(Integer.MIN_VALUE, Integer.MAX_VALUE);
+			}
+
+			testArray(args);
+		});
 	}
 
-	private static Stream<Arguments> dynamicTests() {
-		// <region tests>
-		return Stream.of(
-				//
-				// Artemis Example
-				//
-				arguments(
-						"Artemis Example",
-						new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 },
-						new int[] { 8, 4, 12, 1, 5, 9, 13, 3, 7, 11, 15, 2, 6, 10, 14 }),
-				//
-				// Empty Example
-				//
-				arguments(
-						"Empty Test",
-						new int[] {},
-						new int[] {})
-		// </end-region>
-		);
+	// private static int seedDifference = 0;
+	// private static Stream<Arguments> testArray() {
+	// 	// TODO improve this
+	// 	return Stream.generate(() -> {
+	// 		return arguments((new Random(seed + seedDifference++)).ints(seedDifference, 0, Integer.MAX_VALUE).toArray());
+	// 	}).limit(10);
+	// }
```

Comment:
due to troubles with generics and junit, i also couldn't solve how to make the method `testArray` a ParameterizedTest without falling into conversion errors within the argument stream